app/src/main/java/com/tz/sanga/moviestore/Fragments/Host/HostPresenter.java

```diff
@@ -0,0 +1,92 @@
+package com.tz.sanga.moviestore.Fragments.Host;
+
+import com.tz.sanga.moviestore.BuildConfig;
+import com.tz.sanga.moviestore.Model.API.Connector;
+import com.tz.sanga.moviestore.Model.API.Service;
+import com.tz.sanga.moviestore.Model.Movie;
+import com.tz.sanga.moviestore.Model.MoviesResponse;
+
+import java.util.List;
+
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+
+
+public class HostPresenter {
+    private static final int PAGE_START = 1;
+    private boolean isLoading = false;
+    private boolean isLastPage = false;
+    private int TOTAL_PAGES = 20;
+    private int currentPage = PAGE_START;
+    private HostView hostView;
+    private Service movieService;
+    private int movePreference;
+
+    public HostPresenter(HostView hostView, int movePreference) {
+        this.hostView = hostView;
+        this.movePreference = movePreference;
+    }
+
+    public void getData(){
+        hostView.showLoading();
+        movieService= Connector.getConnector().create(Service.class);
```

Comment:
This can be initialized inly once in the constructor and used in the entire class